lib/Filters/FilePathFilter.cpp

```diff
@@ -55,15 +55,30 @@ bool FilePathFilter::shouldSkip(const std::string &sourceFilePath) const {
     cache[sourceFilePath] = allow;
   }
 
-  return (! cache[sourceFilePath]);
+  return (!cache[sourceFilePath]);
 }
 
-std::string FilePathFilter::name() { return "file path"; }
+std::string FilePathFilter::name() {
+  return "file path";
+}
+
+std::pair<bool, std::string> FilePathFilter::exclude(const std::string &filter) {
 
-void FilePathFilter::exclude(const std::string &filter) {
-  excludeFilters.emplace_back(filter, std::regex::egrep);
+  auto regex_filter = llvm::Regex(filter);
+  std::string error;
+  if (!regex_filter.isValid(error)) {
+    return std::make_pair(false, std::move(error));
+  }
+  excludeFilters.emplace_back(filter);
```

Comment:
Shouldn't it contain `regex_filter`?

Reply:
You are right, I messed this up, because I was trying to push back the string and create the regex each time it will get checked. But making it mutable seems to make more sense here.

Reply:
Based on the CI failure, I think these should be `std::move`d.

Reply:
If I remember correctly move ctor has been deleted. I will have a look.

Reply:
Ok, if that's the case, then we can just construct each regex twice, should be fine.